modules/generator/registry/counter.go

```diff
@@ -147,30 +150,55 @@ func (c *counter) collectMetrics(appender storage.Appender, timeMs int64) (activ
 			endOfLastMinuteMs := getEndOfLastMinuteMs(timeMs)
 			_, err = appender.Append(0, s.labels, endOfLastMinuteMs, 0)
 			if err != nil {
+				c.seriesMtx.RUnlock()
```

Comment:
Do you think we can avoid this manual unlocking? Makes it very easy to have a danging deadlock.